api/src/main/java/io/minio/MinioClient.java

```diff
@@ -5741,12 +5743,16 @@ public void listenBucketNotification(String bucketName, String prefix, String su
       scanner = new Scanner(response.body().charStream());
       scanner.useDelimiter("\n");
       while (scanner.hasNext()) {
```

Comment:
My suggestion is to deprecate this style of API and introduce lazy eval'ed style like
```java
public Iterable<Result<NotificationInfo>> listenBucketNotification(String bucketName, String prefix, String suffix, String[] events)
```

If we stick with current style, the fix is just `while (!continueListening || scanner.hasNext())`

Reply:
A more idiomatic Java style is preferred, so I think lazy evaled style is better and bump the version for breakage

Reply:
@balamurugana  need your input on this . I have implemented the `listenBucketNotification` as below but it waits at `scanner.next()` .Please review the code.
```
public Iterable<Result<NotificationInfo>> listenBucketNotification1(String bucketName, String prefix,
      String suffix, String[] events) {
    return new Iterable<Result<NotificationInfo>>() {

      @Override
      public Iterator<Result<NotificationInfo>> iterator() {
        return new Iterator<Result<NotificationInfo>>() {

          private Result<NotificationInfo> error;
          NotificationInfo ni;
          private boolean completed = false;

          private synchronized void populate() {
            Multimap<String,String> queryParamMap = HashMultimap.create();
            queryParamMap.put("prefix", prefix);
            queryParamMap.put("suffix", suffix);
            for (String event: events) {
              queryParamMap.put("events", event);
            }

            String bodyContent = "";
            Scanner scanner = null;
            HttpResponse response = null;
            ObjectMapper mapper = new ObjectMapper();

            try {
              response = executeReq(Method.GET, getRegion(bucketName),
                bucketName, "", null, queryParamMap, null, 0);
              scanner = new Scanner(response.body().charStream());
              scanner.useDelimiter("\n");
              if (scanner.hasNext()) {
                bodyContent = scanner.next().trim();
                if (!bodyContent.equals("")) {
                  ni = mapper.readValue(bodyContent, NotificationInfo.class);
                }
              }
            } catch (InvalidBucketNameException | NoSuchAlgorithmException | InsufficientDataException | IOException
              | InvalidKeyException | NoResponseException | XmlPullParserException | ErrorResponseException
              | InternalException | InvalidResponseException | RuntimeException e) {
              this.error = new Result<>(null, e);
            }             finally {
              if (response != null) {
                response.body().close();
              }
              if (scanner != null) {
                scanner.close();
              }
            }

          }


          @Override
          public boolean hasNext() {
            if (this.completed) {
              return false;
            }

            if (this.error == null ) {
              populate();
            }

            if (this.ni == null ) {
              populate();
            }

            this.completed = true;
            return false;
          }

          @Override
          public Result<NotificationInfo> next() {
            if (this.completed) {
              throw new NoSuchElementException();
            }
            if (this.error == null ) {
              populate();
            }
            this.completed = true;
            throw new NoSuchElementException();
          }

          @Override
          public void remove() {
            throw new UnsupportedOperationException();
          }
        };
      }
    };
  }
```

Reply:
This looks neat :+1: @sinhaashish unless @balamurugana has any concerns